R/staggregate.R

```diff
@@ -190,6 +212,17 @@ polygon_aggregation <- function(clim_dt, weights_dt, list_names, time_agg){
 #' @export
 staggregate_polynomial <- function(data, overlay_weights, daily_agg, time_agg = "month", degree){
 
+
+  # Stop if they've requested hourly output and but not daily_agg = none
+  if(time_agg == "hour" & daily_agg != "none"){
+    if(daily_agg == "sum"){
+      stop(crayon::red("Cannot output hourly data after aggregating to daily totals. Please specify 'daily_agg = \"none\"' to get hourly output."))
+    }
+    if(daily_agg == "average"){
+      stop(crayon::red("Cannot output hourly data after aggregating to daily averages. Please specify 'daily_agg = \"none\"' to get hourly output."))
+    }
+  }
+
```

Comment:
I wonder if there's a way we can make it so the user only has to specify one of the inputs (daily agg or time_agg) to get hourly data back. Like maybe if time_agg = 'hour' we can set daily_agg to none by default? Which would override any mistakes with having time_agg = 'hour' but daily_agg as sum or average. Maybe it's not a big deal though if we want to leave as is, I just figured it might avoid any complications. Maybe the warning can just say something like: `if(time_agg = 'hour'){ message(crayon::yellow("Daily aggregate is set to none for hourly output") }`

Reply:
Great idea yeah if we can save one person from setting their code running and coming back hours later to an easily fixed error we should

Reply:
Thanks for adding that!